feat: add SAML SSO documentation (#1935)
mendersoftware/mender-docs: 08.Server-integration/08.SAML-Federated-Authentication/docs.md

---
title: SAML Federated Authentication
taxonomy:
    category: docs
---

!!!!! SAML Federated Authentication is only available in the Mender Enterprise plan.
!!!!! See [the Mender features page](https://mender.io/product/features?target=_blank)
!!!!! for an overview of all Mender plans and features.

It is possible to configure a Mender tenant to leverage a SAML-compatible Identity Provider (IdP) to identify and authenticate users. Thanks to this feature, users can log in to Mender using their existing credentials, e.g., Azure/Microsoft, AWS, or any other service compatible with SAML 2.0.

## Configuration

Setting up Mender as a SAML 2.0 Service Provider (SP) requires uploading the SAML metadata from your IdP in the "Organization and billing" settings view.

![SAML configuration](saml_1.png)

As Mender uses the user's email address as a username, please make sure your IdP is configured to use the email addresses as NameID format.
You can verify it by checking the metadata XML you downloaded from your IdP and ensuring that it contains the following line:

```xml
<md:NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress</md:NameIDFormat>
```

> Once the XML metadata file upload is complete, you will obtain the Start URL (login URL), which will initiate the SAML authentication for your user.

Reviewer: I think it would be good to briefly describe Entity ID and ACS URL, because these can be required by, IdP when configuring SAML SSO on IdP side.
Entity ID is the endpoint which provides Mender SP metadata, and usually this one, or the metadata file, is required by IdP.

Author: Thanks @kjaskiewiczz, I added a section about it.

---
title: SAML Federated Authentication
taxonomy:
    category: docs
---

!!!!! SAML Federated Authentication is only available in the Mender Enterprise plan.
!!!!! See [the Mender features page](https://mender.io/product/features?target=_blank)
!!!!! for an overview of all Mender plans and features.

> It is possible to configure a Mender tenant to leverage a SAML-compatible Identity Provider (IdP) to identify and authenticate users. Thanks to this feature, users can log in to Mender using their existing credentials, e.g., Azure/Microsoft, AWS, Google, or any other service compatible with SAML 2.0.

Reviewer 2: ```suggestion
It is possible to configure a Mender tenant to leverage a SAML-compatible Identity Provider (IdP) to identify and authenticate users. Thanks to this feature, users can log in to Mender using their existing credentials, e.g. Azure/Microsoft, AWS, Google, or any other service compatible with SAML 2.0.
```

> It is possible to configure a Mender tenant to leverage a SAML-compatible Identity Provider (IdP) to identify and authenticate users. Thanks to this feature, users can log in to Mender using their existing credentials, e.g., Azure/Microsoft, AWS, Google, or any other service compatible with SAML 2.0.

Reviewer 2: Maybe it's just me, but I find sentences that starts with "Thanks to..." a little weird, it's like there is an implicit "you're welcome" between the lines :upside_down_face:.
```suggestion
It is possible to configure a Mender tenant to leverage a SAML-compatible Identity Provider (IdP) to identify and authenticate users.
Using SAML, users can log in to Mender using existing single sign-on credentials from services compatible with SAML 2.0, like Azure/Microsoft, AWS and Google.```
